packages/neon-api/src/txService/builder.ts

```diff
@@ -0,0 +1,186 @@
+    cosigners.forEach(cosigner => {
+      if (typeof cosigner.account !== "string") {
+        this._cosignerAccounts.push(cosigner.account);
+        newCosingers.push(
+          Object.assign({}, cosigner, {
+            account: reverseHex(cosigner.account.scriptHash)
+          })
+        );
+      } else {
+        this._cosignerAccounts.push(new Account(cosigner.account));
+        newCosingers.push(cosigner as CosignerLike);
+      }
+    });
+
+    this._cosigners.push(...newCosingers);
+    this._transaction.addCosigner(...newCosingers);
+    return this;
+  }
+
+  /**
+   * You can add multiple intents to the transaction
+   * @param intents
+   */
+  public addIntents(...intents: ScriptIntent[]): this {
+    this._transaction.script += createScript(...intents);
+    this._intents.push(...intents);
+    return this;
+  }
+
+  /**
+   * Add an attribute.
+   * @param usage The usage type. Do refer to txAttrUsage enum values for all available options.
+   * @param data The data as hexstring.
+   */
+  public addAttribute(usage: number | string, data: string): this {
+    this._transaction.addAttribute(new TransactionAttribute({ usage, data }));
+    return this;
+  }
+
+  /**
+   * You can just use `sign` function to do this.
+   * Still this is useful when：
+   *   1. Need to accept a cosigner's signature
+   *   2. Need to use verification trigger of a contract address
+   * @param witnesses
+   */
+  public addWitnesses(...witnesses: Witness[]): this {
+    witnesses.forEach(witness => this._transaction.addWitness(witness));
+    return this;
+  }
+
+  public addMultiSig(multisigAccount: Account, ...witnesses: Witness[]) {
+    if (!multisigAccount.isMultiSig) {
+      throw new Error(`${multisigAccount} is not a multi-sig account`);
+    }
+
+    if (
+      ![this._sender, ...this._cosignerAccounts].some(
+        account => account.address === multisigAccount.address
+      )
+    ) {
+      throw new Error(`${multisigAccount} is neither sender nor cosigner`);
+    }
+
+    const multisigWitness = Witness.buildMultiSig(
+      this._transaction.serialize(),
+      witnesses,
+      multisigAccount
+    );
+    return this.addWitnesses(multisigWitness);
+  }
+
+  public sign(account: Account | string): this {
+    this._transaction.sign(account);
+    return this;
+  }
+
+  /**
+   * The transaction must be signed with sender and all other cosigners.
+   * It may be that all signatures cannot be aquired from one client.
+   * This function is to sign the transaction with (sender) aquired accounts.
+   */
+  public signWithAquiredAccounts(): this {
+    [this._sender, ...this._cosignerAccounts].forEach(account => {
+      account.privateKey && this._transaction.sign(account);
+    });
+    return this;
+  }
+
+  public setSystemFee(fee: Fixed8) {
+    this._transaction.systemFee = fee;
+  }
+
+  public setNetworkFee(fee: Fixed8) {
+    this._transaction.networkFee = fee;
+  }
+
+  public export(): Transaction {
```

Comment:
use `build()` and it should produce a new transaction, not just return the reference to the private field.